File: modules/SMM Client Application/src/main/java/net/mtgsaber/smm/client/models/ModpackVersion.java
package net.mtgsaber.smm.client.models;

import java.util.List;
import java.util.UUID;

/**
 *
 */
public record ModpackVersion(
        Modpack modpack,
        String version,
        URI changelog,
        String targetMCVersion,
        URI forgeDLPage,
        List<MCModFile> modFiles,
        List<ConfigFile> configFiles,
        List<PackFile> miscPackFiles,
        MCProfile profile
) {
}

File: SMM-Client-Application/src/main/scala/net/mtgsaber/smm/client/models/ModpackVersion.scala
package net.mtgsaber.smm.client.models

/**
 * Represents a particular version of a Minecraft modpack, including the specific mods and files.
 */
case class ModpackVersion(
  modpack: Modpack, versionID: String, forgeVersion: String,
  modFiles: Map[PackFile, Mod], configFiles: Map[PackFile, Mod],
  otherFiles: List[PackFile], recommendedMemAllocMB: Int
)
